modules/images/webp-uploads/helper.php

```diff
@@ -144,7 +144,10 @@ function webp_uploads_generate_additional_image_source( $attachment_id, $image_s
 
 	// Skip creation of duplicate WebP image if an image file already exists in the directory.
 	if ( file_exists( $destination_file_name ) ) {
-		return new WP_Error( 'webp_image_file_present', __( 'The WebP image already exists.', 'performance-lab' ) );
+		return array(
+			'file'     => wp_basename( $destination_file_name ),
+			'filesize' => wp_filesize( $destination_file_name ),
+		);
```

Comment:
This fixes the problem for regenerating images, but it then reintroduces a bug similar to #395. Let me clarify:
* Before #395 was fixed, a file for an image upload could overwrite a file for a previous image upload.
* After #395 was fixed, no overwrite would happen anymore, but now regeneration will fail for every image.
* With what you have here, regeneration will work again, but then the scenario from #395 would result in the image for the previous upload being used for the new upload, which would be a new bug.

I think we need an if condition here:
* If the `$destination_file_name` belongs to the given `$attachment_id`, then we shouldn't even return anything here - neither the error nor the previous file - since in that case we should actually overwrite the image (e.g. for a regeneration process).
* Otherwise (i.e. `$destination_file_name` is for another attachment), then we should still return the `WP_Error` present here.

Reply:
Yes, it's a tricky issue. If we fix one problem it arises another.

> If the $destination_file_name belongs to the given $attachment_id, then we shouldn't even return anything here - neither the error nor the previous file - since in that case we should actually overwrite the image (e.g. for a regeneration process).

We can't check `$destination_file_name` is from which `$attachment_id` because when we regenerate an existing image `car.jpeg`, we don't have `sources` in metadata. The Regenerate script tries to generate `sources` for the image, but if we don't return an array of file and filesize, then it will not generate a WebP image.

Please check the attachment metadata that I printed before the `file_exists` condition.

![image](https://user-images.githubusercontent.com/10103365/178728643-9831a9cf-e499-4248-8d0b-e9e1a29488b7.png)

If we remove the `file_exists` condition, then only it will work fine for regeneration. We will check that edge case after this release.

Reply:
@mukeshpanchal27 After giving this some further thought, I think the problem here points to a greater issue that we have to consider when designing the feature to regenerate existing images. We need a way to know what the original image files were that were present _before_ starting the regeneration process for an attachment. I think we could use a meta key similar to `_wp_attachment_backup_sizes` to store the image files for the different sizes from right before the regeneration temporarily, until the process has been completed for the image. This will allow us to know whether an existing file belongs to the same attachment or not. cc @jjgrainger 

With that said, current solutions to regenerate existing images do not have that. While that is not great, I would argue, returning an error here is definitely the safer path. We may end up with some content not having WebP files when regenerated, but we absolutely cannot risk mixing up image files from another attachment. So I would say that we should close this PR without merging it. Great job finding that problem though, since it will be crucial for the regenerating images implementation!

Reply:
Yes agreed. Let me close the PR.